internal/internal_task_pollers.go

```diff
@@ -941,6 +947,19 @@ func newGetHistoryPageFunc(
 		} else {
 			h = resp.History
 		}
+
+		size := len(h.Events)
+		// While the SDK is processing a workflow task, the workflow task could timeout and server would start
+		// a new workflow task or the server looses the workflow task if it is a speculative workflow task. In either
+		// case, the new workflow task could have events that are beyond the last event ID that the SDK expects to process.
+		// In such cases, the SDK should return error indicating that the workflow task is stale since the result will not be used.
+		if size > 0 && lastEventID > 0 && 
+			h.Events[size-1].GetEventId() > lastEventID {
+			return nil, nil, fmt.Errorf("history contains events past expected last event ID (%v) "+
```

Comment:
Can you confirm that an error iterating here is treated as task failure (eviction and all of that)?

Reply:
Yes